.github/workflows/run-tests.yml

```diff
@@ -1,19 +1,64 @@
 name: Run-Tests
+
 on:
   push:
     branches: [ main ]
   pull_request:
     branches: [ main ]
 
 jobs:
+  validate-dependencies:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0  # Fetch all history for all branches and tags
+
+      - name: Fetch all history
+        run: git fetch --no-tags --prune --depth=1 origin +refs/heads/*:refs/remotes/origin/*
+
+      - name: Check for changes in dependency files
+        id: check-deps
+        run: |
+          echo "Checking pyproject.toml and poetry.lock for changes..."
+          CHANGES=$(git diff --name-only origin/main origin/feature_ci_auto_deps | grep -E "^(pyproject.toml|poetry.lock)$" || true)
+          echo "Changed files: $CHANGES"
+          if [ ! -z "$CHANGES" ]; then
+            echo "Checking if requirements.txt has also been updated..."
+            if git diff --name-only origin/main origin/feature_ci_auto_deps | grep -q "requirements.txt"; then
+              echo "requirements.txt updated correctly."
```

Comment:
I think you could just have the negative condition instead of both positive and negative. When I run something like this, I assume the positive condition is met, I only care about the result if I get a negative result.

Reply:
This makes sense I think I've handled it correctly in b448ab3f0b64f119aa1645c571f780b41636ff08.